python-threatexchange/threatexchange/signal_type/tests/test_pdq_index2.py

```diff
@@ -0,0 +1,138 @@
+import typing as t
+import numpy as np
+import random
+
+from threatexchange.signal_type.pdq.pdq_index2 import PDQIndex2
+from threatexchange.signal_type.pdq.signal import PdqSignal
+from threatexchange.signal_type.pdq.pdq_utils import convert_pdq_strings_to_ndarray
+
+SAMPLE_HASHES = [PdqSignal.get_random_signal() for _ in range(100)]
+
+
+def _brute_force_match(
+    base: t.List[str], query: str, threshold: int = 32
+) -> t.Set[int]:
+    matches = set()
+    query_arr = convert_pdq_strings_to_ndarray([query])[0]
+
+    for i, base_hash in enumerate(base):
+        base_arr = convert_pdq_strings_to_ndarray([base_hash])[0]
+        distance = np.count_nonzero(query_arr != base_arr)
+        if distance <= threshold:
+            matches.add(i)
+    return matches
+
+
+def _generate_random_hash_with_distance(hash: str, distance: int) -> str:
+    if len(hash) != 64 or not all(c in "0123456789abcdef" for c in hash.lower()):
+        raise ValueError("Hash must be a 64-character hexadecimal string")
+    if distance < 0 or distance > 256:
+        raise ValueError("Distance must be between 0 and 256")
+
+    hash_bits = bin(int(hash, 16))[2:].zfill(256)  # Convert hash to binary
+    bits = list(hash_bits)
+    positions = random.sample(
+        range(256), distance
+    )  # Randomly select unique positions to flip
+    for pos in positions:
+        bits[pos] = "0" if bits[pos] == "1" else "1"  # Flip selected bit positions
+    modified_hash = hex(int("".join(bits), 2))[2:].zfill(64)  # Convert back to hex
+
+    return modified_hash
+
+
+def test_pdq_index():
+    # Make sure base_hashes and query_hashes have at least 100 similar hashes
+    base_hashes = SAMPLE_HASHES + [PdqSignal.get_random_signal() for _ in range(1000)]
+    query_hashes = SAMPLE_HASHES + [PdqSignal.get_random_signal() for _ in range(10000)]
```

Comment:
Why add SAMPLE_HASHES here? If you are trying to guarantee matches, you could generate less query hashes and copy the banked hashes in